import_export/forms.py

```diff
@@ -5,7 +5,25 @@
 from django.utils.translation import gettext_lazy as _
 
 
-class ImportForm(forms.Form):
+class ImportExportFormBase(forms.Form):
+    resource = forms.ChoiceField(
+        label=_('Resource'),
+        choices=(),
+        required=False,
+    )
+
+    def __init__(self, resources=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        try:
+            resource_choices = []
+            for i, resource in enumerate(resources):
+                resource_choices.append((i, resource.get_display_name()))
+            self.fields['resource'].choices = resource_choices
+        except TypeError:
+            del self.fields['resource']
```

Comment:
I think it is a bit of a code smell to rely on the `TypeError` here.  Perhaps `isinstance(resources, list)` might be better.

Reply:
I didn't want to discriminate tuple, so I ended up with this solution. The correct solution is `hasattr(resources, '__getitem__')`.